e2e/cluster/cluster.go

```diff
@@ -191,23 +192,28 @@ func NewTestCluster(in *Input) *Output {
 			CreateRegularUser(in, node)
 		}
 	}
-	if in.WithProxy {
-		CreateProxy(in)
-	}
-	return &Output{
+	out := &Output{
 		T:       in.T,
 		Nodes:   nodes,
 		network: in.network,
 		id:      in.id,
 	}
+	if in.WithProxy {
+		out.Proxy = CreateProxy(in)
```

Comment:
clusters that set up a proxy failed when calling `Destroy` since we weren't setting the proxy node name here